amr-wind/physics/SyntheticTurbulence.cpp

```diff
@@ -0,0 +1,574 @@
+    << " m/s; Dir = " << wind_direction * 180.0 / pi
+    << " deg; type = " << mean_wind_type << std::endl;
+}
+
+void SyntheticTurbulence::initialize_fields(
+    int // level
+    ,
+    const amrex::Geometry& // geom)
+)
+{
+    //TODO: Figure out what goes here
+}
+
+void SyntheticTurbulence::pre_advance_work()
+{
+  if (m_is_init)
+    initialize();
+
+  update();
+}
+
+void SyntheticTurbulence::initialize()
+{
+  // Convert current time to an equivalent length based on the reference
+  // velocity to determine the position within the turbulence grid
+  const amrex::Real curTime = m_time.new_time() - m_time_offset;
+  const amrex::Real eqivLen = m_wind_profile->reference_velocity() * curTime;
+  int il, ir;
+  get_lr_indices(m_turb_grid, 0, eqivLen, il, ir);
+  load_turb_plane_data(m_turb_filename, m_turb_grid, il, ir);
+
+  m_is_init = false;
+}
+
+void SyntheticTurbulence::update()
+{
+  // Convert current time to an equivalent length based on the reference
+  // velocity to determine the position within the turbulence grid
+  const amrex::Real cur_time = m_time.new_time() - m_time_offset;
+  const amrex::Real eqiv_len = m_wind_profile->reference_velocity() * cur_time;
+
+  InterpWeights weights;
+  get_lr_indices(m_turb_grid, 0, eqiv_len,
+                 weights.il, weights.ir, weights.xl, weights.xr);
+
+  // Check if we need to refresh the planes
+  if (weights.il != m_turb_grid.ileft)
+    load_turb_plane_data(m_turb_filename, m_turb_grid, weights.il, weights.ir);
+
+  auto& repo = m_turb_force.repo();
+  auto& geom_vec = repo.mesh().Geom();
+
+  const int sdir = (*m_wind_profile).shear_dir();
+
+  const int nlevels = repo.num_active_levels();
+  for (int lev=0; lev < nlevels; ++lev) {
+      const auto& geom = geom_vec[lev];
+      const auto& problo = geom.ProbLoArray();
+
+      const amrex::Real dx = geom.CellSize()[0];
+      const amrex::Real dy = geom.CellSize()[1];
+      const amrex::Real dz = geom.CellSize()[2];
+
+      for (amrex::MFIter mfi(m_turb_force(lev)); mfi.isValid(); ++mfi) {
+          const auto& bx = mfi.tilebox();
+          const auto& turb_force_arr = m_turb_force(lev).array(mfi);
+          const auto& rho_arr = m_density(lev).array(mfi);
+
+          amrex::ParallelFor(
+              bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+
+            vs::Vector xyz_g;
+            vs::Vector xyz_l;
+            // velocity in local frame
+            vs::Vector vel_l;
+            // velocity in global frame
+            vs::Vector vel_g;
+
+            xyz_g[0] = problo[0] + (i + 0.5) * dx;
+            xyz_g[1] = problo[1] + (j + 0.5) * dy;
+            xyz_g[2] = problo[2] + (k + 0.5) * dz;
+
+            // Transform position vector from global inertial
+            // reference frame to local reference frame attached to
+            // the turbulence grid.
+            xyz_l = m_turb_grid.tr_mat & ( xyz_g - m_turb_grid.origin);
```

Comment:
```suggestion
            xyz_l = trmat & ( xyz_g - origin);
```

No `m_*` class instance variable can appear inside `amrex::ParallelFor`